### PR TITLE
add util for loss spike save and decode.

### PR DESCRIPTION
**Pull request by haikuotiankong1212**
### What changes were proposed in this pull request?

针对loss尖刺的记录和解析提供了一个工具

### Why are the changes needed?

为了更多训练的人来使用

### CODE REVIEW

File: atorch/atorch/utils/loss_spike_utils.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+
+class LossSpikeBase:
+    def __init__(self, loss_spike_save_dir, sample_data_paths, each_sample_len, min_iter, min_loss,
+                 loss_info_splitter='\t', loss_sample_str_splitter=','):
+        """
+        init params
+        Args:
+            loss_spike_save_dir: str, The directory where loss spike files are stored
+            sample_data_paths: 样本存储的路径信息, 可以是用户自定义的结构，如名称+路径的tuple list:
```
Comment (workingloong) [inline atorch/atorch/utils/loss_spike_utils.py:19]: It is better to use English annotations.

**Comment by workingloong** [comment]
You need to format your commits to pass the test of atorch-pre-commit.

File: atorch/atorch/utils/loss_spike_utils.py (hunk 1 above)
Comment (haikuotiankong1212) [reply atorch/atorch/utils/loss_spike_utils.py]: Done.

File: atorch/docs/README-LOSS-SPIKE-UTIL.md
Comment (skydoorkai) [inline atorch/docs/README-LOSS-SPIKE-UTIL.md:1]: Use English for document writing and with readable format.

File: atorch/atorch/utils/loss_spike_utils.py [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+
+    @staticmethod
+    def get_data_file_len(fpath, dtype):
+        with open(fpath) as f:
+            f.seek(0, 2)
+            return f.tell() // dtype.itemsize
+
+
+class TokenLossSpike(LossSpikeBase):
+    def save_loss(self, file_name, cur_loss, cur_iter, losses_str, sample_infos_str):
```
Comment (skydoorkai) [inline atorch/atorch/utils/loss_spike_utils.py:57]: What is the relationship between cur_loss and losses_str, or cur_iter and sample_infos_str?
What do losses_str and sample_infos_str mean in model training ?

File: atorch/atorch/utils/loss_spike_utils.py [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+        max_loss = losses[idx]
+        sample_with_max_loss = sample_infos[idx]
+        ds, data = self.fetch(sample_with_max_loss)
+        if ds is None:
+            return None, None, None
+        if tokenizer is not None:
+            data = tokenizer.decode(data)
+        return ds, data, max_loss
+
+    def fetch(self, each_sample_info):
```
Comment (skydoorkai) [inline atorch/atorch/utils/loss_spike_utils.py:139]: Since fetch is user-defined method, then either:
define it in base class as abstract method.
or 
the class instance initialization should have a parameter (fetch_func) , which is provided by user.

**Comment by BalaBalaYi** [comment]
Does this PR still need updates and merging? If so, please reply. This PR will be closed by the end of the month if there is no response. Thanks a lot.

**Comment by workingloong** [comment]
Atorch has been removed from the repo to https://github.com/intelligent-machine-learning/atorch. You can feel free to submit a new PR in Atorch.
